src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1110,10 +1110,19 @@ int shim::resetDevice(xclResetKind kind)
 
     dev_fini();
 
+    unsigned int timer = xrt_core::config::get_device_offline_timer();
+    std::chrono::time_point<std::chrono::system_clock> start, end;
+    start = std::chrono::system_clock::now();
     while (dev_offline) {
         std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Magic constant and no comments.  This logic, regardless of magic or not, must be commented; it is not obvious.